transformer_engine/pytorch/optimizers/fused_adam.py

```diff
@@ -259,6 +259,10 @@ def _apply_scale(self, state_name, unscaled_state, scaled_state, scale):
             scale (torch.Tensor): A FP32 tensor representing the scaling factor.
         """
         assert unscaled_state.dtype == torch.float32
+        if scaled_state.dtype == torch.bfloat16:
+            scaled_state.copy_(unscaled_state.bfloat16())
+            return
```

Comment:
It's uncomfortable that FP16 involves scaling while BF16 does not (see similar complaint at https://github.com/NVIDIA/TransformerEngine/pull/1465). Even if this does the optimal behavior, it's not obvious to the user that changing the dtype would also change the compute. Perhaps in a future PR we should add an option for state scaling so that we disentangle the dtype from the algorithm logic.